File: abc083-b/python/main.py
def digit_sum(n):
    return sum(int(d) for d in str(n))


range_max, digit_sum_min, digit_sum_max = map(int, input().split())

sum = sum(
    x for x in range(1, range_max + 1) if digit_sum_min <= digit_sum(x) <= digit_sum_max
)

print(sum)

File: abc083-b/js/main.js
function Main(input) {
    const [rangeMax, digitSumMin, digitSumMax] = input.split(" ").map(Number);

    const digitSum = (n) => {
        return n.toString().split("").map(Number).reduce((acc, cur) => acc + cur, 0);
    }

    const sum = Array.from({ length: rangeMax }, (_, i) => i + 1)
        .filter((n) => {
            const sum = digitSum(n);
            return sum >= digitSumMin && sum <= digitSumMax;
        }).reduce((acc, cur) => acc + cur, 0);

    console.log(sum);
}

Main(require("fs").readFileSync("/dev/stdin", "utf8"));
